lib/rex/proto/ssh/connection.rb

```diff
@@ -0,0 +1,346 @@
+# -*- coding: binary -*-
+require 'rex/proto/ssh/hrr_rb_ssh'
+
+module Rex
+module Proto
+module Ssh
+
+##
+# Whitelist-based access control scaffold
+##
+module AccessControlList
+
+  #
+  # Add permitted access control entry to access control list
+  # Create ACL if it does not yet exist
+  #
+  # @param host [String] Host/hostname for which to grant access
+  # @param port [Integer] Port for which to grant access
+  # @param bind [TrueClass,FalseClass] Whether this ACE is for servers
+  #
+  def permit=(host, port, bind = false)
+    @acl ||= { bind:[], connect:[] }
+    unless permit?(host, port, bind)
+      @acl[ bind ? :bind : :connect ] << "#{host}:#{port}"
+    end
+  end
+
+  #
+  # Delete permitted access control entry from access control list
+  #
+  # @param host [String] Host/hostname for which to grant access
+  # @param port [Integer] Port for which to grant access
+  # @param bind [TrueClass,FalseClass] Whether this ACE is for servers
+  #
+  def deny=(host, port, bind = false)
+    @acl[ bind ? :bind : :connect ].select! do |ent|
+      ent != "#{host}:#{port}"
+    end if @acl
+  end
+
+  #
+  # Check if access control entry exists in access control list
+  #
+  # @param host [String] Host/hostname for which to check access
+  # @param port [Integer] Port for which to check access
+  # @param bind [TrueClass,FalseClass] Whether this ACE is for servers
+  #
+  # @return [TrueClass,FalseClass] Permission boolean for access
+  def permit?(host, port, bind = false)
+    @acl and ["#{host}:#{port}", "*:*", "#{host}:*", "*:#{port}"].any? do |m|
+      @acl[ bind ? :bind : :connect ].include?(m)
+    end
+  end
+end
+
+##
+# Encapsulation of Connection constructor for Rex use
+# Provides ACLs for port forwarding and client (io) access hooks
+##
+class Connection < ::HrrRbSsh::Connection
+  include AccessControlList
+  def self.default_options
+    noneauth = HrrRbSsh::Authentication::Authenticator.new { |context| true }
+    return {
+      'authentication_none_authenticator' => noneauth,
+      'authentication_password_authenticator' => noneauth,
+      'authentication_publickey_authenticator' => noneauth,
+      'authentication_keyboard_interactive_authenticator' => noneauth,
+      'local_version' => 'SSH-2.0-RexProtoSsh'
+    }
+  end
+  #
+  # Create new Connection from an IO and options set, pull trans
+  # and auth from options if present, create from options set otherwise.
+  #
+  # Creates a default empty handler set for channel requests.
+  #
+  # @param io [IO] Socket, FD, or abstraction on which to build Connection
+  # @param options [Hash] Options for constructing Connection components
+  #
+  # @return [Rex::Proto::Ssh::Connection] a new connection object
+  def initialize(io = nil, options = self.default_options, context = {})
+    def_handler = HrrRbSsh::Connection::RequestHandler.new {|c| }
+    @cfd_handlers = {
+      'connection_channel_request_pty_req'       => def_handler,
```

Comment:
Unlike `Authentication::Authenticator`, `Connection::RequestHandler` is not required in `options` when the handler does nothing for each request. The below is an example handler code in the HrrRbSsh library.
```ruby
class Shell < RequestType 
  ...
  def self.run proc_chain, username, io, variables, message, options, session
    ...
    handler = options.fetch('connection_channel_request_shell', RequestHandler.new {})
```
My apologies if you write this to show explicitly the handlers to do nothing.